test/Microsoft.AspNetCore.Server.Kestrel.Performance/KestrelHttpParserRequestHeaders.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions.Internal;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Performance
+{
+    [Config(typeof(CoreConfig))]
+
+    public class KestrelHttpParserRequestHeaders : IHttpHeadersHandler
+    {
+        private const int InnerLoopCount = 512;
+
+        private ReadableBuffer _singleRequestHeaderBuffer = ReadableBuffer.Create(RequestParsingData._singleRequestHeader);
+        private ReadableBuffer _liveaspnetRequestHeadersBuffer = ReadableBuffer.Create(RequestParsingData._liveaspnetRequestHeaders);
+
+        private KestrelHttpParser _parser = new KestrelHttpParser(new NullLog());
+
+        [Benchmark(Baseline = true, OperationsPerInvoke = InnerLoopCount)]
+        public void ParseSingleRequestHeader()
+        {
+            for (var i = 0; i < InnerLoopCount; i++)
+            {
+                ParseData(_singleRequestHeaderBuffer);
+            }
+        }
+
+        [Benchmark(OperationsPerInvoke = InnerLoopCount)]
+        public void ParseLiveAspNet()
+        {
+            for (var i = 0; i < InnerLoopCount; i++)
+            {
+                ParseData(_liveaspnetRequestHeadersBuffer);
+            }
+        }
+
+        private void ParseData(ReadableBuffer buffer)
+        {
+            _parser.ParseHeaders(this, buffer, out var consumed, out var examined, out var consumedBytes);
+        }
+
+        public void OnHeader(Span<byte> name, Span<byte> value)
+        {
+        }
+
+        private class NullLog : IKestrelTrace
```

Comment:
It should be fine to pass null if you are not expecting reject request to be called.